modules/core/03-connection/keeper/verify.go

```diff
@@ -366,6 +366,127 @@ func (k Keeper) VerifyNextSequenceRecv(
 	return nil
 }
 
+// VerifyChannelUpgradeError verifies a proof of the provided upgrade error receipt.
+func (k Keeper) VerifyChannelUpgradeError(
+	ctx sdk.Context,
+	connection exported.ConnectionI,
+	proofHeight exported.Height,
+	proofErrorReceipt []byte,
+	portID,
+	channelID string,
+	errorReceipt channeltypes.ErrorReceipt,
+) error {
+	clientID := connection.GetClientID()
+	clientState, clientStore, err := k.getClientStateAndVerificationStore(ctx, clientID)
+	if err != nil {
+		return err
+	}
+
+	if status := k.clientKeeper.GetClientStatus(ctx, clientState, clientID); status != exported.Active {
+		return errorsmod.Wrapf(clienttypes.ErrClientNotActive, "client (%s) status is %s", clientID, status)
+	}
+
+	merklePath := commitmenttypes.NewMerklePath(host.ChannelUpgradeErrorPath(portID, channelID))
+	merklePath, err = commitmenttypes.ApplyPrefix(connection.GetCounterparty().GetPrefix(), merklePath)
+	if err != nil {
+		return err
+	}
+
+	bz, err := k.cdc.Marshal(&errorReceipt)
+	if err != nil {
+		return err
+	}
+
+	if err := clientState.VerifyMembership(
+		ctx, clientStore, k.cdc, proofHeight,
+		0, 0, // skip delay period checks for non-packet processing verification
+		proofErrorReceipt, merklePath, bz,
+	); err != nil {
+		return errorsmod.Wrapf(err, "failed upgrade error receipt verification for client (%s)", clientID)
+	}
+
+	return nil
+}
+
+// VerifyChannelUpgradeErrorAbsence verifies a proof of the absence of a
+// channel upgrade error.
+func (k Keeper) VerifyChannelUpgradeErrorAbsence(
```

Comment:
I don't think this is used anymore?

Reply:
Just checked and you're right!

Reply:
https://github.com/cosmos/ibc-go/issues/5477

Reply:
Addressed in https://github.com/cosmos/ibc-go/pull/5532